llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -5889,10 +5889,17 @@ void TargetLowering::ComputeConstraintToUse(AsmOperandInfo &OpInfo,
     for (const unsigned E = G.size();
          BestIdx < E && (G[BestIdx].second == TargetLowering::C_Other ||
                          G[BestIdx].second == TargetLowering::C_Immediate);
-         ++BestIdx)
+         ++BestIdx) {
       if (lowerImmediateIfPossible(G[BestIdx], Op, DAG, *this))
         break;
+      // If we're out of constraints, just pick the first one.
+      if (BestIdx + 1 == E) {
+        BestIdx = 0;
+        break;
+      }
+    }
 
+    assert(BestIdx < G.size() && "bad index for best constraint");
```

Comment:
llvm SmallVector::operator[] already has this assert.

Reply:
done in https://github.com/llvm/llvm-project/pull/67494/commits/88e6d3a105b103c0601a804c4932afcb0cb42e04